### PR TITLE
import: add new sources to import from the web

### PR DESCRIPTION
**Pull request by benerken**
* Adds Library of Congress catalog.
* Adds DNB catalog.
* Adds SLSP catalog.
* Adds UGent catalog.
* Adds KULeuven catalog.

Co-Authored-by: Benoit Erken <erken.benoit@gmail.com>
Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on `rero-ils#<xx>`'s PR(s):

* rero/rero-ils#2394

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/api/import-source-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  constructor(
+    private _httpClient: HttpClient
+  ) { }
+
+  /**
+   * Get the configuration entrypoint for external sources
+   * @return: an Observable on external sources list
+   */
+  getSources(): Observable<Array<ExternalSourceSetting>> {
```
Comment (Garfield-fr) [inline projects/admin/src/app/api/import-source-api.service.ts:42]: `Observable<ExternalSourceSetting[]>`

File: projects/admin/src/app/routes/import-documents-route.ts [hunk 2]
```diff
@@ -45,58 +62,51 @@ export class ImportDocumentsRoute extends BaseRoute implements RouteInterface {
...
-            },
-            allowEmptySearch: false
-          }
-        ]
+        types: []
       }
     };
+    this._routeToolService.injector.get(ImportSourceApiService)
+      .getSources()
+      .subscribe((sources: Array<ExternalSourceSetting>) => {
```
Comment (Garfield-fr) [inline projects/admin/src/app/routes/import-documents-route.ts:77]: `sources: ExternalSourceSetting[]`

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/api/import-source-api.service.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/api/import-source-api.service.ts:42]: @Garfield-fr Tsssss :D It's just the same write in a different way :D

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts [hunk 3]
```diff
@@ -297,13 +297,13 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
    */
    private _extractAndFormatQueryParams(identifier: any): string {
     const query = [];
-    const type = identifier.type.replace(/:/g, '\\:');
-    const value = identifier.value.replace(/:/g, '\\:');
-    query.push(`identifiedBy.type:${type}`);
-    query.push(`identifiedBy.value:${value}`);
+    /* const type = identifier.type.replace(/:/g, '\\:');
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts:300]: To be removed?

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts [hunk 4]
```diff
@@ -297,13 +297,13 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
...
+    const value = identifier.value.replace(/:/g, '\\:'); */
+    query.push(`identifiedBy.type:"${identifier.type}"`);
+    query.push(`identifiedBy.value:"${identifier.value}"`);
     if (identifier.source) {
-      const source = identifier.source.replace(/:/g, '\\:');
-      query.push(`identifiedBy.source:${source}`);
+      /* const source = identifier.source.replace(/:/g, '\\:'); */
```
Comment (jma) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts:305]: to be removed?

File: projects/admin/src/app/routes/import-documents-route.ts [hunk 5]
```diff
@@ -45,58 +62,51 @@ export class ImportDocumentsRoute extends BaseRoute implements RouteInterface {
...
+        sources.forEach((source: ExternalSourceSetting) => {
+          const sourceConfig = this._loadConfigSource(source);
+          config.data.types.push(sourceConfig);
+        });
+      });
+    return config;
```
Comment (jma) [inline projects/admin/src/app/routes/import-documents-route.ts:83]: Seems wired as the config is returned immediately and the types is completed later. Returned a observable can be a good option.

File: projects/admin/src/app/routes/import-documents-route.ts (hunk 2 above)
Comment (jma) [inline projects/admin/src/app/routes/import-documents-route.ts:77]: I would use  a pipe, map.

**Review by jma — APPROVED** [review]

File: projects/admin/src/app/routes/import-documents-route.ts (hunk 5 above)
Comment (benerken) [reply projects/admin/src/app/routes/import-documents-route.ts:83]: Implement a observable requires to refactor a lot a stuff ! 
The current solution is the best one according to the time available.

File: projects/admin/src/app/routes/import-documents-route.ts (hunk 2 above)
Comment (benerken) [reply projects/admin/src/app/routes/import-documents-route.ts:77]: The current solution is the best one according to the time available.

**Pull request merged**